test/integration/sonobuoy_integration_test.go

```diff
@@ -723,27 +689,99 @@ func TestPluginComplexCmds_LocalGolden(t *testing.T) {
 				output.WriteTo(&allOutput)
 			}
 
-			binaryVersion := mustRunSonobuoyCommand(t, "version --short")
-			binaryVer := strings.TrimSpace(binaryVersion.String())
+			checkFileMatchesOrUpdate(t, allOutput.String(), tc.expectFile, tmpDir)
+		})
+	}
+}
 
-			outString := strings.ReplaceAll(allOutput.String(), binaryVer, "*STATIC_FOR_TESTING*")
-			outString = strings.ReplaceAll(outString, tmpDir, "*STATIC_FOR_TESTING*")
-			r := regexp.MustCompile("time=\".*?\"")
-			outString = r.ReplaceAllString(outString, `time="STATIC_TIME_FOR_TESTING"`)
+// TestPluginLoading_LocalGolden uses gen/goldenfile flow as the easiest/best way
+// to test that plugins are loaded from the correct places.
+func TestPluginLoading_LocalGolden(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
 
-			if *update {
-				if err := os.WriteFile(tc.expectFile, []byte(outString), 0666); err != nil {
-					t.Fatalf("Failed to update goldenfile: %v", err)
-				}
-			} else {
-				fileData, err := ioutil.ReadFile(tc.expectFile)
-				if err != nil {
-					t.Fatalf("Failed to read golden file %v: %v", tc.expectFile, err)
-				}
-				if diff := pretty.Compare(string(fileData), outString); diff != "" {
-					t.Errorf("Expected manifest to equal goldenfile: %v but got diff:\n\n%v", tc.expectFile, diff)
-				}
-			}
-		})
+	installedPluginFile := "./testdata/plugin-loading-installed.golden"
+	localPluginFile := "./testdata/plugin-loading-local.golden"
+
+	tmpDir, err := ioutil.TempDir("", "sonobuoy_plugin_test_*")
+	if err != nil {
+		t.Fatal("Failed to create tmp dir")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	os.Setenv("SONOBUOY_DIR", tmpDir)
+	os.Setenv("SONOBUOY_ALL_FEATURES", "true")
+
+	_, e := runSonobuoyCommandWithContext(ctx, t, "gen -p hello-world.yaml --kubernetes-version=v123.456.789")
+	if e == nil {
+		t.Fatal("Expected a failure since no plugin was installed but got none")
+	}
+
+	_ = mustRunSonobuoyCommandWithContext(ctx, t, "plugin install hello-world.yaml ./testdata/plugins/good/hello-world.yaml")
+	output := mustRunSonobuoyCommandWithContext(ctx, t, "gen -p hello-world.yaml --kubernetes-version=v123.456.789")
+	checkFileMatchesOrUpdate(t, output.String(), installedPluginFile, tmpDir)
+
+	_ = mustRunSonobuoyCommandWithContext(ctx, t, "plugin uninstall hello-world.yaml")
+	_, e = runSonobuoyCommandWithContext(ctx, t, "gen -p hello-world.yaml --kubernetes-version=v123.456.789")
+	if e == nil {
+		t.Fatal("Expected a failure since no plugin was installed but got none")
+	}
+
+	// Copy file to pwd
+	input, err := ioutil.ReadFile("./testdata/plugins/good/hello-world.yaml")
+	if err != nil {
+		t.Fatalf("Failed to read plugin to test pwd loading")
+	}
+
+	err = ioutil.WriteFile("hello-world.yaml", bytes.Replace(input, []byte("foo.com"), []byte("localfile.com"), -1), 0644)
```

Comment:
This is the point of this test partially. We need to differentiate which plugin was loaded so they need to be unique. The one that is in the pwd will have this source instead fo the foo.com value.